File: internal/rules/no_console/no_console_test.go
package no_console

import (
	"testing"

	"github.com/web-infra-dev/rslint/internal/plugins/typescript/rules/fixtures"
	"github.com/web-infra-dev/rslint/internal/rule_tester"
)

func TestNoConsoleRule(t *testing.T) {
	rule_tester.RunRuleTester(
		fixtures.GetRootDir(),
		"tsconfig.json",
		t,
		&NoConsoleRule,
		[]rule_tester.ValidTestCase{
			{
				Code:    `console.warn("test");`,
				Options: map[string]interface{}{"allow": []interface{}{"warn"}},
			},
			{
				Code:    `console.error("test");`,
				Options: map[string]interface{}{"allow": []interface{}{"error"}},
			},
			{
				Code:    `console.warn("test"); console.error("test");`,
				Options: map[string]interface{}{"allow": []interface{}{"warn", "error"}},
			},
			{Code: `var x = { console: 1 };`},
			// Shadowed console should not be reported
			{Code: `function f(console: any) { console.log("x"); }`},
		},
		[]rule_tester.InvalidTestCase{
			{
				Code: `console.log("test");`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "unexpected", Line: 1, Column: 1},
				},
			},
			{
				Code: `console.warn("test");`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "unexpected", Line: 1, Column: 1},
				},
			},
			{
				Code: `console.error("test");`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "unexpected", Line: 1, Column: 1},
				},
			},
			{
				Code: `console.info("test");`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "unexpected", Line: 1, Column: 1},
				},
			},
			{
				Code:    `console.log("test");`,
				Options: map[string]interface{}{"allow": []interface{}{"warn"}},
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "unexpected", Line: 1, Column: 1},
				},
			},
			// Computed property access: console["log"]
			{
				Code: `console["log"]("test");`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "unexpected", Line: 1, Column: 1},
				},
			},
			// Chained member access: console.log.bind should still report console.log
			{
				Code: `console.log.bind(null)("test");`,
				Errors: []rule_tester.InvalidTestCaseError{
					{MessageId: "unexpected", Line: 1, Column: 1},
				},
			},
		},
	)
}

File: packages/rslint-test-tools/tests/eslint/rules/no-console.test.ts
import { RuleTester } from '../rule-tester';

const ruleTester = new RuleTester();

ruleTester.run('no-console', {
  valid: [
    {
      code: 'console.warn("test");',
      options: { allow: ['warn'] },
    },
    {
      code: 'console.error("test");',
      options: { allow: ['error'] },
    },
    {
      code: 'console.warn("test"); console.error("test");',
      options: { allow: ['warn', 'error'] },
    },
    'var x = { console: 1 };',
    // Shadowed console should not be reported
    'function f(console: any) { console.log("x"); }',
  ],
  invalid: [
    {
      code: 'console.log("test");',
      errors: [{ messageId: 'unexpected' }],
    },
    {
      code: 'console.warn("test");',
      errors: [{ messageId: 'unexpected' }],
    },
    {
      code: 'console.error("test");',
      errors: [{ messageId: 'unexpected' }],
    },
    {
      code: 'console.info("test");',
      errors: [{ messageId: 'unexpected' }],
    },
    {
      code: 'console.log("test");',
      options: { allow: ['warn'] },
      errors: [{ messageId: 'unexpected' }],
    },
    // Computed property access
    {
      code: 'console["log"]("test");',
      errors: [{ messageId: 'unexpected' }],
    },
    // Chained member access
    {
      code: 'console.log.bind(null)("test");',
      errors: [{ messageId: 'unexpected' }],
    },
  ],
});
